### PR TITLE
feat(CPH): add a new datasource of server flavors

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cph' TESTARGS='-run=TestAccDatasourceServerFlavors_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cph -v -run=TestAccDatasourceServerFlavors_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceServerFlavors_basic
=== PAUSE TestAccDatasourceServerFlavors_basic
=== CONT  TestAccDatasourceServerFlavors_basic
--- PASS: TestAccDatasourceServerFlavors_basic (7.08s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cph       7.144s
```

### CODE REVIEW

File: huaweicloud/services/cph/data_source_huaweicloud_cph_server_flavors.go [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+		}
+
+		rst = append(rst, v)
+	}
+	return rst
+}
+
+func buildListFlavorsQueryParams(d *schema.ResourceData) string {
+	res := ""
+	if v, ok := d.GetOk("type"); ok {
```
Comment (ShiChangkuo) [inline huaweicloud/services/cph/data_source_huaweicloud_cph_server_flavors.go:258]: if type is 0, ok will be false

File: huaweicloud/services/cph/data_source_huaweicloud_cph_server_flavors.go [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+	var mErr *multierror.Error
+
+	// listFlavors: Query the list of CPH server flavors
+	var (
+		listFlavorsHttpUrl = "v1/{project_id}/cloud-phone/server-models"
+		listFlavorsProduct = "cph"
+	)
+	listFlavorsClient, err := cfg.NewServiceClient(listFlavorsProduct, region)
+	if err != nil {
+		return diag.Errorf("error creating ServerFlavors Client: %s", err)
```
Comment (ShiChangkuo) [inline huaweicloud/services/cph/data_source_huaweicloud_cph_server_flavors.go]:
```suggestion
		return diag.Errorf("error creating CPH Client: %s", err)
```

can we use the uppercase of `listFlavorsProduct` in template?

File: huaweicloud/services/cph/data_source_huaweicloud_cph_server_flavors.go (hunk 1 above)
Comment (hmbig2) [reply huaweicloud/services/cph/data_source_huaweicloud_cph_server_flavors.go:258]: ok,change to string

File: huaweicloud/services/cph/data_source_huaweicloud_cph_server_flavors.go (hunk 2 above)
Comment (hmbig2) [reply huaweicloud/services/cph/data_source_huaweicloud_cph_server_flavors.go]: ok, fix the template

**Pull request merged**